src/backend/marsha/core/api.py

```diff
@@ -76,24 +74,26 @@ def update_state(request):
     # Update the object targeted by the "object_id" and "resource_id"
     model = apps.get_model(app_label="core", model_name=key_elements["model_name"])
 
-    update_parameters = {}
-    if serializer.validated_data["state"] == defaults.READY:
-        update_parameters["uploaded_on"] = key_elements["uploaded_on"]
-        if hasattr(model, "extension"):
-            update_parameters["extension"] = key_elements.get("extension")
-
-    updated = model.objects.filter(id=key_elements["object_id"]).update(
-        **{
-            # Only update `uploaded_on` if the upload was actually successful
-            **update_parameters,
-            "upload_state": serializer.validated_data["state"],
-        }
-    )
-
-    if updated:
-        return Response({"success": True})
+    extra_parameters = serializer.validated_data["extraParameters"]
+    if serializer.validated_data["state"] == defaults.READY and hasattr(
+        model, "extension"
+    ):
+        extra_parameters["extension"] = key_elements.get("extension")
+
+    try:
+        object_instance = model.objects.get(id=key_elements["object_id"])
+        object_instance.update_upload_state(
+            upload_state=serializer.validated_data["state"],
+            uploaded_on=key_elements.get("uploaded_on")
+            if serializer.validated_data["state"] == defaults.READY
+            else None,
+            **extra_parameters,
+        )
+        object_instance.save()
+    except model.DoesNotExist:
+        return Response({"success": False}, status=404)
```

Comment:
only this line should be in the try, the rest should be in an "else" clause. 
```suggestion
    try:
        object_instance = model.objects.get(id=key_elements["object_id"])
    except model.DoesNotExist:
        return Response({"success": False}, status=404)
    else:
        object_instance.update_upload_state(
            upload_state=serializer.validated_data["state"],
            uploaded_on=key_elements.get("uploaded_on")
            if serializer.validated_data["state"] == defaults.READY
            else None,
            **extra_parameters,
        )
        object_instance.save()
```

Reply:
The `else` is useless IMO because if the exception is caught the response is immediately return.

Reply:
ok but the exception might happen when calling `update_upload_state` and this is not the situation you expected so you wouldn't want to run the code in the expect...

Reply:
```
try:
        object_instance = model.objects.get(id=key_elements["object_id"])
    except model.DoesNotExist:
        return Response({"success": False}, status=404)

    object_instance.update_upload_state(
        upload_state=serializer.validated_data["state"],
        uploaded_on=key_elements.get("uploaded_on")
        if serializer.validated_data["state"] == defaults.READY
        else None,
        **extra_parameters,
    )
```

I don't run it in the except